File: notification/src/main/java/com/ezbuy/notification/repository/NotificationCategoryRepository.java
package com.ezbuy.notification.repository;

import org.springframework.data.r2dbc.repository.Query;
import org.springframework.data.r2dbc.repository.R2dbcRepository;

import com.ezbuy.notification.model.NotificationCategory;
import com.ezbuy.notification.repository.query.NotificationCategoryQuery;

import reactor.core.publisher.Mono;

public interface NotificationCategoryRepository extends R2dbcRepository<NotificationCategory, String> {
    Mono<NotificationCategory> findById(String id);

    @Query(NotificationCategoryQuery.findCategoryIdByType)
    Mono<String> findCategoryIdByType(String categoryType);
}

File: notification-service/src/main/kotlin/com/ezbuy/notificationservice/repository/NotificationCategoryRepository.kt
package com.ezbuy.notificationservice.repository

import com.ezbuy.notificationservice.repository.query.NotificationCategoryQuery
import com.ezbuy.notificationmodel.model.NotificationCategory
import org.springframework.data.r2dbc.repository.Query
import org.springframework.data.r2dbc.repository.R2dbcRepository
import reactor.core.publisher.Mono

interface NotificationCategoryRepository : R2dbcRepository<NotificationCategory, String> {

    override fun findById(id: String): Mono<NotificationCategory>

    @Query(NotificationCategoryQuery.FIND_CATEGORY_ID_BY_TYPE)
    fun findCategoryIdByType(categoryType: String): Mono<String>
}
